### ISSUE TITLE
Ubuntu Touch does not run on unmodified Halium

### ISSUE THREAD

**Issue by UniversalSuperBox** [ubports/ubuntu-touch#404]
## The Problem

Right now, as can be seen at our [porting information](https://docs.ubports.com/en/latest/porting/introduction.html), porters need to do all of the work to port Halium and then modify it again to run Ubuntu Touch. The modifications will likely introduce more problems into the port that need to be fixed.

This is a problem because it:

* greatly increases the time that a porter must spend on their port to run Ubuntu Touch
* forces us to continue to maintain projects that we shouldn't. We helped create Halium so that we wouldn't need to do this.

## How we fix it

In a nutshell, these are the changes we need to make to run on an unmodified Halium base. Some will require upstreaming changes to Halium. I will add more as I discover them.

Items will be checked off this list when they are shipping in their respective projects. I will link relevant pull requests or issues.

- [ ] Halium does not provide a recovery that can be used to update Ubuntu Touch
- [ ] System-image does not handle the Android system.img and rootfs being separate
- [ ] Halium configured kernels do not provide AppArmor

And finally, once that is done...

- [ ] Ship any changes in an Ubuntu Touch image
- [ ] Ship a Halium-based device image from our system-image server
  
 **Update** 2017-01-13: After spending many hours of work making Ubuntu Touch compatible with hybris-boot, we've decided that it's probably a better idea to cut our losses and try to make ubports-boot as friendly to Halium as possible. To start, we've merged a PR that means that ubports-boot and hybris-boot will not conflict when placed in a source tree together. I've updated this issue to reflect our new decision.

**Comment by UniversalSuperBox** [comment]
This issue was a symptom of https://github.com/ubports/ubuntu-touch/issues/446, which was fixed via https://github.com/ubports/ubuntu-push/commit/1921122388353e8e840e84dc37663060351fdd04.

**Comment by UniversalSuperBox** [comment]
It's time for the...

## Halium update!

Here's the current status of Ubuntu Touch's integration with Halium:

* [halium-boot](https://github.com/halium/halium-boot), a boot image based on Debian and Ubuntu Touch's initramfs images combined, is coming along nicely. However, the default mountpoints currently come from Ubuntu Touch. This is good for us, not so good for other Halium OS's. A configuration system needs to be put in place before it can really be accepted as standard. Interesting ways to follow this are [Remarks on Halium-boot for LuneOS](https://github.com/Halium/projectmanagement/issues/61) and [Change from hybris-boot to halium-boot](https://github.com/Halium/projectmanagement/issues/55)
* Ports are rather difficult when practically every Android 7.1 device doesn't work. #494 highlights a huge problem that we have with all of our graphical applications on devices that ship with CodeAurora Forum's Android for MSM. It also highlights a couple of ways to fix it, which we would love some help on.
* I attempted to run Ubuntu Touch on the Nexus 5X. [It didn't go very well](https://i.imgur.com/zROaZ1d.mp4). Yes, that's the dash crashing in the background. Yes, that's the keyboard. The problems experienced included a segfault when trying to call `localtime()` in certain contexts. I wasn't sure of the next steps and neither was anyone who I asked, so I've shelved the port for now. It was a bit too much of a time sink when we have bigger problems ([16.04 backlog](https://github.com/ubports/ubuntu-touch/milestone/7), [OTA-4](https://github.com/ubports/ubuntu-touch/milestone/8)) at the moment.

**Comment by UniversalSuperBox** [comment]
I don't believe we'll be able to get this in prior to OTA-4. I vote to punt it to OTA-5.

**Comment by slp** [comment]
@UniversalSuperBox I'm pretty sure those segfaults in localtime() are related to #677
